src/libs/actions/Policy.ts

```diff
@@ -941,7 +1009,7 @@ function updateWorkspaceCustomUnitAndRate(policyID, currentCustomUnit, newCustom
                     [currentCustomUnit.customUnitID]: {
                         customUnitID: currentCustomUnit.customUnitID,
                         rates: {
-                            [currentCustomUnit.rates.customUnitRateID]: {
+                            [newCustomUnit.rates.customUnitRateID]: {
```

Comment:
Why it was changed?

Reply:
Hi @VickyStash great catch! This is actually something that I changed before Tymek took over the PR. I'm pretty sure this is a mistake someone made before, that's why it's corrected now (TS helped me notice it). 

This is due to two reasons:
1. If you take a look at OnyxUpdates located here (Optimistic & SuccesData), you can see that in OptimisticData we are creating a rate in `customUnits[newCustomUnit.customUnitID].rates` with ID `[newCustomUnit.rates.customUnitRateID]`, this is also the only rate created for `[newCustomUnit.customUnitID]` custom unit. After that happens once the request is successful we want to update the `pendingAction` to `null` there's no rate with the ID of `currentCustomUnit.rates.customUnitRateID` in our newly created unit.
2. I have checked it to make sure by console logging `currentCustomUnit`, and it turns out that `currentCustomUnit.rates.customUnitRateID` is `undefined`, so it doesn't exist, making this line useless

Hope this clears up the confusion. Additionally, I ran a few tests and everything seems to be working just fine. Your comment however proved that the if statement located above was incorrect so we'll add a fix for that. Thank you!